Feature/158 name registry register application

ITStar10 opened the pull request.

ITStar10 committed:
[VDA-Name-Registry] Support application regisering

Work on progress

ITStar10 committed:
[VDA-Name-Registry] Work in progress

- Support registering an application

ITStar10 committed:
Hardhat configuration for Amoy updated

ITStar10 committed:
[VDA-Name-Registry] Work on progress

ITStar10 committed:
[NameRegistryContract] v1.0.0

- Create `NameRegistryV2` contract that support registering an application
- Update the folder structure of the `contract` directory
- Update the test code

ITStar10 committed:
[NameRegistryV2] 1.0.0

- Added `getVersion()` function

ITStar10 committed:
[VDA-Name-Registry] v1.0.0

- Added 2 functions
	`isValidSuffix(...)`
	`getSuffixList()`

tahpot commented on VDA-Name-Registry/contracts/v2/NameRegistryV2.sol:
```diff
@@ -0,0 +1,548 @@
+    /**
+     * @dev See {INameRegistryV2}
+     */
+    function deRegisterApp(
```
`deregisterApp`

ITStar10 replied:
Updated

tahpot commented on VDA-Name-Registry/contracts/v2/NameRegistryV2.sol:
```diff
@@ -0,0 +1,548 @@
+        _didApps[did].remove(_appName);
+        delete _didAppMetaData[did][_appName];
+
+        emit DeRegisterApp(did, _ownerName, _appName);
```
`emit deregisterApp`

ITStar10 replied:
All the other event names are started with the uppercase letter.
For example, the event for the `registerApp()` function is `RegisterApp`.
How about to name the event for the `deregisterApp()` function as `Deregisterapp`?

tahpot replied:
Yes `DeregisterApp`

ITStar10 replied:
Updated

tahpot commented on VDA-Name-Registry/contracts/v2/NameRegistryV2.sol:
```diff
@@ -0,0 +1,548 @@
+    /**
+     * @dev See {INameRegistryV2}
+     */
+    function enableAppRegister(bool isEnabled) external virtual payable override onlyOwner {
```
Need `disableAppRegister`

ITStar10 replied:
As we've already deployed the `NameRegistry` contract on the testnet and mainnet, we should upgrade the current contracts. Here, the registering app feature is now disabled in the original `NameRegistry` contract.
After we upgrade the contract, we should set the fee for registering app and set the token address. 
And then, we should enable the app registering by call this `enableAppRegister(true)` function.

So, I think the function name of `enableAppRegister()` might be suitable.

tahpot replied:
I agree with the name, I'm saying we should have an extra method to disable app registration. We may want to do that in the future.

ITStar10 replied:
Once we called `enableAppRegister(false)`, it would disable the registering app.

tahpot replied:
Oh I see, sorry I missed the boolean.

In that case, make it `isAppRegisterEnabled()`

ITStar10 replied:
There is already `isAppRegisterEnabled()` function that returns the boolean.
![image](https://github.com/user-attachments/assets/b54d93c5-5fbd-4f6e-86ca-21b29159431f)

tahpot replied:
Haha ok.

Let's call it `setAppRegisterEnabled()`

ITStar10 replied:
Updated

tahpot reviewed (approved):
Approved with minor cahnges

ITStar10 committed:
[XPRewardContract] v0.1.1

- Removed gas price for the `Amoy` test net in the hardhat configuration
- Testnet token contract address in the deployment script

ITStar10 committed:
[NameRegistry] v1.0.0

- Updates on review

ITStar10 committed:
[NameRegistry] v1.0.0

- Added `script/upgrade_manual.ts`
